### ISSUE TITLE
debian 8 jessie - bridge problems

### ISSUE THREAD

**Issue by bertenvdb** [docker/docker#18793]
I'm trying to install docker-engine on a fresh debian 8 (jessie). I followed the instructions listed [here](https://docs.docker.com/engine/installation/debian/#debian-jessie-80-64-bit).
Unfortunately, docker-engine fails to install using apt, and after having wasted so many hours trying to solve this I have absolutely no idea what to do next. I also tried to install version 1.9.0-0~jessie, but this generates the same errors.

uname -a:

    Linux seven 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1+deb8u3 (2015-08-04) x86_64 GNU/Linux

This is what happens when I try to install:

    berten@seven:~$ sudo apt-get install docker-engine
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    The following extra packages will be installed:
      aufs-tools cgroupfs-mount git git-man libapparmor1 liberror-perl libnih-dbus1 libnih1 makedev mountall patch
      plymouth
    Suggested packages:
      git-daemon-run git-daemon-sysvinit git-doc git-el git-email git-gui gitk gitweb git-arch git-cvs git-mediawiki
      git-svn ed diffutils-doc desktop-base plymouth-themes
    The following NEW packages will be installed:
      aufs-tools cgroupfs-mount docker-engine git git-man libapparmor1 liberror-perl libnih-dbus1 libnih1 makedev
      mountall patch plymouth
    0 upgraded, 13 newly installed, 0 to remove and 3 not upgraded.
    Need to get 13.0 MB of archives.
    After this operation, 62.6 MB of additional disk space will be used.
    Do you want to continue? [Y/n] 
    Get:1 http://ftp.belnet.be/debian/ jessie/main makedev all 2.3.1-93 [42.6 kB]
    Get:2 http://ftp.belnet.be/debian/ jessie/main plymouth amd64 0.9.0-9 [189 kB]
    Get:3 https://apt.dockerproject.org/repo/ debian-jessie/main docker-engine amd64 1.9.1-0~jessie [7,344 kB]
    Get:4 http://ftp.belnet.be/debian/ jessie/main libnih1 amd64 1.0.3-4.3 [127 kB]
    Get:5 http://ftp.belnet.be/debian/ jessie/main libnih-dbus1 amd64 1.0.3-4.3 [97.1 kB]
    Get:6 http://ftp.belnet.be/debian/ jessie/main mountall amd64 2.54 [68.3 kB]
    Get:7 http://ftp.belnet.be/debian/ jessie/main patch amd64 2.7.5-1 [109 kB]
    Get:8 http://ftp.belnet.be/debian/ jessie/main aufs-tools amd64 1:3.2+20130722-1.1 [92.9 kB]
    Get:9 http://ftp.belnet.be/debian/ jessie/main cgroupfs-mount all 1.1 [4,572 B]
    Get:10 http://ftp.belnet.be/debian/ jessie/main liberror-perl all 0.17-1.1 [22.4 kB]
    Get:11 http://ftp.belnet.be/debian/ jessie/main git-man all 1:2.1.4-2.1 [1,266 kB]
    Get:12 http://ftp.belnet.be/debian/ jessie/main git amd64 1:2.1.4-2.1 [3,624 kB]
    Get:13 http://ftp.belnet.be/debian/ jessie/main libapparmor1 amd64 2.9.0-3 [60.3 kB]
    Fetched 13.0 MB in 1s (7,787 kB/s)        
    Selecting previously unselected package makedev.
    (Reading database ... 86075 files and directories currently installed.)
    Preparing to unpack .../makedev_2.3.1-93_all.deb ...
    Unpacking makedev (2.3.1-93) ...
    Selecting previously unselected package plymouth.
    Preparing to unpack .../plymouth_0.9.0-9_amd64.deb ...
    Unpacking plymouth (0.9.0-9) ...
    Selecting previously unselected package libnih1.
    Preparing to unpack .../libnih1_1.0.3-4.3_amd64.deb ...
    Unpacking libnih1 (1.0.3-4.3) ...
    Selecting previously unselected package libnih-dbus1.
    Preparing to unpack .../libnih-dbus1_1.0.3-4.3_amd64.deb ...
    Unpacking libnih-dbus1 (1.0.3-4.3) ...
    Selecting previously unselected package mountall.
    Preparing to unpack .../mountall_2.54_amd64.deb ...
    Unpacking mountall (2.54) ...
    Selecting previously unselected package patch.
    Preparing to unpack .../patch_2.7.5-1_amd64.deb ...
    Unpacking patch (2.7.5-1) ...
    Selecting previously unselected package aufs-tools.
    Preparing to unpack .../aufs-tools_1%3a3.2+20130722-1.1_amd64.deb ...
    Unpacking aufs-tools (1:3.2+20130722-1.1) ...
    Selecting previously unselected package cgroupfs-mount.
    Preparing to unpack .../cgroupfs-mount_1.1_all.deb ...
    Unpacking cgroupfs-mount (1.1) ...
    Selecting previously unselected package liberror-perl.
    Preparing to unpack .../liberror-perl_0.17-1.1_all.deb ...
    Unpacking liberror-perl (0.17-1.1) ...
    Selecting previously unselected package git-man.
    Preparing to unpack .../git-man_1%3a2.1.4-2.1_all.deb ...
    Unpacking git-man (1:2.1.4-2.1) ...
    Selecting previously unselected package git.
    Preparing to unpack .../git_1%3a2.1.4-2.1_amd64.deb ...
    Unpacking git (1:2.1.4-2.1) ...
    Selecting previously unselected package libapparmor1:amd64.
    Preparing to unpack .../libapparmor1_2.9.0-3_amd64.deb ...
    Unpacking libapparmor1:amd64 (2.9.0-3) ...
    Selecting previously unselected package docker-engine.
    Preparing to unpack .../docker-engine_1.9.1-0~jessie_amd64.deb ...
    Unpacking docker-engine (1.9.1-0~jessie) ...
    Processing triggers for man-db (2.7.0.2-5) ...
    Processing triggers for systemd (215-17+deb8u2) ...
    Processing triggers for dbus (1.8.20-0+deb8u1) ...
    Setting up makedev (2.3.1-93) ...
    /run/udev or .udevdb or .udev presence implies active udev.  Aborting MAKEDEV invocation.
    /run/udev or .udevdb or .udev presence implies active udev.  Aborting MAKEDEV invocation.
    /run/udev or .udevdb or .udev presence implies active udev.  Aborting MAKEDEV invocation.
    /run/udev or .udevdb or .udev presence implies active udev.  Aborting MAKEDEV invocation.
    Setting up plymouth (0.9.0-9) ...
    update-initramfs: deferring update (trigger activated)
    update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
    update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
    Setting up libnih1 (1.0.3-4.3) ...
    Setting up libnih-dbus1 (1.0.3-4.3) ...
    Setting up mountall (2.54) ...
    Setting up patch (2.7.5-1) ...
    Setting up aufs-tools (1:3.2+20130722-1.1) ...
    Setting up cgroupfs-mount (1.1) ...
    Setting up liberror-perl (0.17-1.1) ...
    Setting up git-man (1:2.1.4-2.1) ...
    Setting up git (1:2.1.4-2.1) ...
    Setting up libapparmor1:amd64 (2.9.0-3) ...
    Setting up docker-engine (1.9.1-0~jessie) ...
    Job for docker.service failed. See 'systemctl status docker.service' and 'journalctl -xn' for details.
    invoke-rc.d: initscript docker, action "start" failed.
    dpkg: error processing package docker-engine (--configure):
     subprocess installed post-installation script returned error exit status 1
    Processing triggers for systemd (215-17+deb8u2) ...
    Processing triggers for initramfs-tools (0.120) ...
    update-initramfs: Generating /boot/initrd.img-3.16.0-4-amd64
    Processing triggers for libc-bin (2.19-18+deb8u1) ...
    Processing triggers for dbus (1.8.20-0+deb8u1) ...
    Errors were encountered while processing:
     docker-engine
    E: Sub-process /usr/bin/dpkg returned an error code (1)

systemctl status docker.service

    ● docker.service - Docker Application Container Engine
       Loaded: loaded (/lib/systemd/system/docker.service; enabled)
       Active: failed (Result: exit-code) since Sat 2015-12-19 17:12:33 CET; 14min ago
         Docs: https://docs.docker.com
      Process: 4770 ExecStart=/usr/bin/docker daemon -H fd:// (code=exited, status=1/FAILURE)
     Main PID: 4770 (code=exited, status=1/FAILURE)

apt-cache policy docker-engine

    docker-engine:
      Installed: 1.9.1-0~jessie
      Candidate: 1.9.1-0~jessie
      Version table:
         1.9.1-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
            100 /var/lib/dpkg/status
         1.9.0-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.8.3-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.8.2-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.8.1-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.8.0-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.7.1-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.7.0-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.6.2-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.6.1-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.6.0-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages
         1.5.0-0~jessie 0
            500 https://apt.dockerproject.org/repo/ debian-jessie/main amd64 Packages

**Comment by itoffshore** [comment]
This is still an issue on `Debian Jessie` with the standard & `backported 4.2` kernel. The default bridge `docker0` does not get created & specifying a custom bridge does not work either:
```
# docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 12:59:02 UTC 2015
 OS/Arch:      linux/amd64
Error response from daemon: 404 page not found

# docker info
Error response from daemon: 404 page not found

# uname -a
Linux toshiba-laptop 4.2.0-0.bpo.1-amd64 #1 SMP Debian 4.2.6-3~bpo8+2 (2015-12-14) x86_64 GNU/Linux

# systemctl status -l docker.service
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled)
   Active: failed (Result: exit-code) since Sat 2015-12-26 20:45:52 GMT; 14s ago
     Docs: https://docs.docker.com
  Process: 7080 ExecStart=/usr/bin/docker daemon -H fd:// (code=exited, status=1/FAILURE)
 Main PID: 7080 (code=exited, status=1/FAILURE)

Dec 26 20:45:51 toshiba-laptop docker[7080]: time="2015-12-26T20:45:51.796176788Z" level=info msg="API listen on /var/run/docker.sock"
Dec 26 20:45:51 toshiba-laptop docker[7080]: time="2015-12-26T20:45:51.891776096Z" level=warning msg="Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man docker` to refer to dm.thinpooldev section."
Dec 26 20:45:51 toshiba-laptop docker[7080]: time="2015-12-26T20:45:51.966620110Z" level=info msg="[graphdriver] using prior storage driver \"devicemapper\""
Dec 26 20:45:51 toshiba-laptop docker[7080]: time="2015-12-26T20:45:51.974443382Z" level=info msg="Firewalld running: false"
Dec 26 20:45:52 toshiba-laptop docker[7080]: time="2015-12-26T20:45:52.144308069Z" level=fatal msg="Error starting daemon: Error initializing network controller: Error creating default \"bridge\" network: failed to parse pool request for address space \"LocalDefault\" pool \"\" subpool \"\": could not find an available predefined network"
Dec 26 20:45:52 toshiba-laptop systemd[1]: docker.service: main process exited, code=exited, status=1/FAILURE
Dec 26 20:45:52 toshiba-laptop systemd[1]: Failed to start Docker Application Container Engine.
Dec 26 20:45:52 toshiba-laptop systemd[1]: Unit docker.service entered failed state.

# docker network create docker0
Error response from daemon: 404 page not found
```

**Comment by thaJeztah** [comment]
ping @mavenugo any suggestions here?

**Comment by itoffshore** [comment]
I've updated my comment above as I managed to get `docker` working by manually creating `docker0`.

There does seem to be a problem with networking as the `libvirt` bridge `virbr0` cannot be shared.

**Comment by itoffshore** [comment]
Editing `/lib/systemd/system/docker.service ` directly to:
```
ExecStart=/usr/bin/docker daemon -b virbr0 -g /data/docker -H fd://
```
Has no effect:
```
# service docker status 
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled)
   Active: active (running) since Sat 2015-12-26 22:06:31 GMT; 2min 30s ago
     Docs: https://docs.docker.com
 Main PID: 3818 (docker)
   CGroup: /system.slice/docker.service
           └─3818 /usr/bin/docker daemon -H fd://
```

**Comment by itoffshore** [comment]
With kernel `3.16` (which has the `AUFS` kernel module)

Setting `$DOCKER_OPTS` in `/etc/default/docker` has no effect but `docker` starts ok if the `docker0` bridge is created manually. To move the `docker` directory I needed to symlink `/var/lib/docker`.

With the `backported 4.2` kernel (**without** the `AUFS` kernel module)

Setting `$DOCKER_OPTS` works but `docker` does not start. It uses `device-mapper` as the backing store & I noticed the `metadata` files increased in size to `100GB` until the partition ran out of space.
